### PR TITLE
Search users

### PR DESCRIPTION
**Pull request by Stuart-Wilcox**
- search for users based on username -> looks for matching query in username. ie `query='wil'` will catch the username `'swilcox7'`
- ex `http://localhost:8080/api/user/search?query=swilcox7`
- updated swagger docs to reflect the change

### CODE REVIEW

File: src/models/user.ts [hunk 1]
```diff
@@ -49,6 +49,10 @@ export class User {
         return new User(userData.username, userData.password, userData.email, userData.fname, userData.lname);
     }
 
+    public static async search(username: string){
+      return await knex('User').select('fname', 'username').where('username', 'like', `%${username}%`);
```
Comment (AgostonSzepessy) [inline src/models/user.ts]: We should return `lname` as well

**Review by AgostonSzepessy — DISMISSED** [review]

File: src/models/user.ts (hunk 1 above)
Comment (Stuart-Wilcox) [reply src/models/user.ts]: Latest commit has the requested changes

File: swagger.json [hunk 2]
```diff
@@ -506,6 +506,33 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully performed search",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "username": { "type": "string" },
+                "fname": { "type": "string" }
```
Comment (AgostonSzepessy) [inline swagger.json]: We should add `lname` here too

**Review by AgostonSzepessy — CHANGES_REQUESTED** [review]

**Review by AgostonSzepessy — APPROVED** [review]
LGTM

**Pull request merged**